lib/Dialect/Arc/Transforms/PrintCostModel.cpp

```diff
@@ -0,0 +1,60 @@
+//===- DummyAnalysisTester.cpp --------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This is a dummy pass to test the cost model results it doesn't do any thing.
+// just walks over the ops to compute some statistics.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/Arc/ArcCostModel.h"
+#include "circt/Dialect/Arc/ArcPasses.h"
+#include "circt/Dialect/HW/HWOps.h"
+#include "mlir/IR/MLIRContext.h"
+#include "mlir/Pass/Pass.h"
+
+#define DEBUG_TYPE "arc-print-cost-model"
+
+namespace circt {
+namespace arc {
+#define GEN_PASS_DEF_PRINTCOSTMODEL
+#include "circt/Dialect/Arc/ArcPasses.h.inc"
+} // namespace arc
+} // namespace circt
+
+using namespace circt;
+using namespace arc;
+
+namespace {
+struct PrintCostModelPass
+    : public arc::impl::PrintCostModelBase<PrintCostModelPass> {
+  void runOnOperation() override;
+
+  OperationCosts statVars;
+};
+} // namespace
+
+void PrintCostModelPass::runOnOperation() {
+  for (auto moduleOp : getOperation().getOps<hw::HWModuleOp>()) {
+    ArcCostModel arcCostModel;
+    moduleOp.walk([&](Operation *op) {
+      OperationCosts curOpCosts = arcCostModel.getCost(op);
+      statVars += curOpCosts;
+    });
+  }
```

Comment:
This might over-count operations, since `walk` will also walk into the body of vectorize ops and count each of the ops separately, but then also call `getCost` on the vectorize op itself which will count stuff in the body again. If you add `HWModuleOp` to the list of ops that `computeOperationCosts` handles as regions->blocks->ops counting, you can do:
```suggestion
  OperationCosts statVars; // no need to carry this in the pass struct
  ArcCostModel arcCostModel; // want to keep the cached results around
  for (auto moduleOp : getOperation().getOps<hw::HWModuleOp>()) {
    statVars += arcCostModel.getCost(moduleOp);
  }
```